docs/source/howto/data.rst

```diff
@@ -249,11 +247,693 @@ The node class has various methods to set these attributes, such as :py:`~aiida.
+                    },
+                    {
+                        'cls':Dict,
+                        'with_outgoing':PwCalculation
+                    }
+                ]
+            }
+
+            # Shorter version:
+
+            qh3 = {
+                'path':[
+                    Dict,
+                    PwCalculation,
+                    TrajectoryData,
+                ]
+            }
+
+*   what to ``project``: Determing which columns the query will return::
+
+        queryhelp = {
+            'path':[PwCalculation],
+            'project':{
+                PwCalculation:['user_id', 'id'],
+            }
+        }
+
+    If you are using JSONB columns, you can also project a value stored inside the json::
+
+        queryhelp = {
+            'path':[
+                PwCalculation,
+                StructureData,
+            ],
+            'project':{
+                PwCalculation:['state', 'id'],
+            }
+        }
+
+    Returns the state and the id of all instances of ``PwCalculation`` where a structures is linked as output of a relax-calculation.
+    The strings that you pass have to be name of the columns.
+    If you pass an asterisk ('*'), the query will return the instance of the AiidaClass.
+
+
+*   the ``filters``:
+    What if you want not every structure, but only the ones that were added after a certain time (say last 4 days) and have an id higher than 50::
+
+        from aiida.common import timezone
+        from datetime import timedelta
+
+        queryhelp = {
+            'path':[
+                {'cls':PwCalculation}, # PwCalculation with structure as output
+                {'cls':StructureData}
+            ],
+            'filters':{
+                StructureData:{
+                    'ctime':{'>':  timezone.now() - timedelta(days=4)},
+                    'id':{'>': 50}
+                }
+            }
+        }
+
+If you want to include filters and projections on links between nodes, you will have to add these to filters and projections in the queryhelp.
+Let's take an example that we had and add a few filters on the link::
+
+    queryhelp = {
+        'path':[
+            {'cls':PwCalculation, 'tag':'relax'}, # PwCalculation with structure as output
+            {'cls':StructureData, 'tag':'structure'}
+        ],
+        'filters':{
+            'structure':{
+                'id':{'>': 50}
+            },
+            'relax--structure':{
+                'label':{'like':'output_%'},
+            }
+        },
+        'project':{
+            'relax--structure':['label'],
+            'structure':['label'],
+            'relax':['label', 'uuid'],
+        }
+    }
+
+Notice that the tag for the link, by default, is the tag of the two connecting nodes delimited by two dashes '--' and the order DOES matter.
+
+If you dislike that way to tag the link, you can choose the tag for the edge in the path when definining the entity to join using ``edge_tag``::
```

Comment:
I'm not sure that "disliking" a proposed best practice provides sufficient motivation.
```suggestion
Alternatively, you can choose the tag for the edge in the path when defining the entity to join using ``edge_tag``::
```